azure_monitor/src/azure_monitor/auto_collection/__init__.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+from opentelemetry.metrics import LabelSet, Meter
+
+from azure_monitor.auto_collection.dependency_metrics import DependencyMetrics
+from azure_monitor.auto_collection.performance_metrics import (
+    PerformanceMetrics,
+)
+from azure_monitor.auto_collection.request_metrics import RequestMetrics
+from azure_monitor.utils import PeriodicTask
+
+__all__ = [
+    "AutoCollection",
+    "DependencyMetrics",
+    "RequestMetrics",
+    "PerformanceMetrics",
+]
+
+
+class AutoCollection:
+    def __init__(
+        self, meter: Meter, label_set: LabelSet, collection_interval: int = 60
+    ):
+        self._performance_metrics = PerformanceMetrics(meter, label_set)
+        self._dependency_metrics = DependencyMetrics(meter, label_set)
+        self._request_metrics = RequestMetrics(meter, label_set)
+
+        self._collect_task = PeriodicTask(
+            interval=collection_interval, function=self.collect
```

Comment:
Should we really be having a separate worker thread for standard metrics?

Reply:
We don't have any other worker thread, and this need to be periodically retrieved so I don't see how else we can do this, any specific idea you have in mind?

Reply:
This leads me to believe that autocollection might have to happen on the sdk rather than in the exporter. It might need to be a separate `azure` owned wrapper around the OT sdk. We should bring this up in the SIG/standup.